src/core/thread/mesh_forwarder.cpp

```diff
@@ -1561,12 +1580,7 @@ void MeshForwarder::ClearReassemblyList(void)
     for (Message &message : mReassemblyList)
     {
         LogMessage(kMessageReassemblyDrop, message, kErrorNoFrameReceived);
-
-        if (message.GetType() == Message::kTypeIp6)
-        {
-            mIpCounters.mRxFailure++;
-        }
-
+        mCounters.UpdateOnDrop(message);
```

Comment:
Side note, it might be useful to separate out counters for different kinds of drop.

Reply:
Exactly. I was thinking of the same :)